### PR TITLE
Read Excel file

### PR DESCRIPTION
**Pull request by rajatbatra0803**
Read Excel file and store all cell value in object

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.idea/
```
Comment (vaibhavjain3) [inline .gitignore]: Why this file has been checked-in?

File: src/main/java/com/erstudio/ERStudioReleaseHelper.java [hunk 2]
```diff
@@ -1,21 +1,37 @@
...
  */
 public class ERStudioReleaseHelper {
 
-	/**
-	 * Launches ERStudio Helper Application
-	 * @param args
-	 */
-	public static void main(String[] args) {
-		// TODO Auto-generated method stub
+    private ChangeTSVersionNumber changeTSVersionNumber;
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERStudioReleaseHelper.java]: This is main launcher file.
Nothing but main function will be there in this file to launch application.

File: src/main/java/com/erstudio/ERStudioReleaseHelper.java [hunk 3]
```diff
@@ -1,21 +1,37 @@
...
+     * @param args
+     */
+    public static void main(String[] args) {
+        // TODO Auto-generated method stub
+        ERStudioReleaseHelper erStudioReleaseHelper = new ERStudioReleaseHelper();
+        erStudioReleaseHelper.readExcel();
+    }
 
+    // function to be triggered from UI button
+    public void readExcel() {
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERStudioReleaseHelper.java]: Entire Read write excel operation has to be moved into separate file.

File: src/main/java/com/erstudio/tsversionchange/ChangeTSVersionNumber.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        this.newVersion = newVersion;
+        this.GUID = GUID;
+        this.upgradeType = upgradeType;
+    }
+
+    public void readExcel() throws IOException {
+        ReadExcel readExcel = new ReadExcel();
+        readExcel.readExcel(this.filePath);
+    }
+}
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/tsversionchange/ChangeTSVersionNumber.java:16]: Why not getters setters?

File: src/main/java/com/erstudio/tsversionchange/ReadExcel.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+        // iterating over Rows and Columns
+        sheet.forEach(row -> {
+            //ignore the first header row
+            if (row.getRowNum() != 0) {
+                Iterator<Cell> cellIterator = row.cellIterator();
+                ExcelModel excelModel = new ExcelModel();
+                while (cellIterator.hasNext()) {
+                    Cell cell1 = cellIterator.next();
+                    //serial number
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/tsversionchange/ReadExcel.java]: Either add comments to all cells or don't add to any.
I prefer, add to all of them.

File: src/main/java/com/erstudio/tsversionchange/model/ExcelModel.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * @author rajat.batra
+ * @description contains data of excel file
+ */
+public class ExcelModel {
+    String filepath;
+    String fullVersion;
+    String patchVersion;
+    String StringToReplaceFormat;
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/tsversionchange/model/ExcelModel.java]: Why variable name is starting with Capital S?

File: src/main/java/com/erstudio/tsversionchange/ReadExcel.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+                    excelModel.setFullVersion(dataFormatter.formatCellValue(cell3));
+                    Cell cell4 = cellIterator.next();
+                    excelModel.setPatchVersion(dataFormatter.formatCellValue(cell4));
+                    Cell cell5 = cellIterator.next();
+                    excelModel.setStringToReplaceFormat(dataFormatter.formatCellValue(cell5));
+                    fileList.add(excelModel);
+                }
+            }
+        });
+        workbook.close();
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/tsversionchange/ReadExcel.java]: try catch finally.

File: src/main/java/com/erstudio/tsversionchange/ChangeTSVersionNumber.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+    public ChangeTSVersionNumber(String filePath, String oldVersion, String newVersion, String GUID, int upgradeType) {
+        this.filePath = filePath;
+        this.oldVersion = oldVersion;
+        this.newVersion = newVersion;
+        this.GUID = GUID;
+        this.upgradeType = upgradeType;
+    }
+
+    public void readExcel() throws IOException {
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/tsversionchange/ChangeTSVersionNumber.java]: Is this is a entity class or a controller class.
What is readExcel function doing in entity class?

**Review by vaibhavjain3 — CHANGES_REQUESTED** [review]

File: src/main/java/com/erstudio/tsversionchange/model/ExcelModel.java (hunk 6 above)
Comment (vaibhavjain3) [reply src/main/java/com/erstudio/tsversionchange/model/ExcelModel.java]: Also,
Make this variable as an object - versionFormat. Create another class with members like major, minor....

File: .gitignore (hunk 1 above)
Comment (rajatbatra0803) [reply .gitignore]: Don't we need .gitignore file to be checked in ?

Comment (vaibhavjain3) [reply .gitignore]: Let's study the use of .gitignore file and post here.

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java [hunk 9]
```diff
@@ -4,7 +4,11 @@
  * @author akshit.arora
  * for the UI Skeleton
  */
+
 import main.java.com.erstudio.constants.Constants;
+import main.java.com.erstudio.tsversionchange.ChangeTSVersionNumber;
+import main.java.com.erstudio.tsversionchange.model.VersionInputModel;
+
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java]: Why Extra line.

File: src/main/java/com/erstudio/tsversionchange/model/ExcelModel.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * @author rajat.batra
+ * @description contains data of excel file
+ */
+public class ExcelModel {
+    String filepath;
+    String fullVersion;
+    String patchVersion;
+    String stringToReplaceFormat;
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/tsversionchange/model/ExcelModel.java]: I believe I asked for creation of new file here. Please check older comments.

**Review by vaibhavjain3 — CHANGES_REQUESTED** [review]

File: CodingGuidelines.md [hunk 11]
```diff
@@ -1,19 +1,17 @@
-Code Check-in Process
-1. Fork Main Repository in local User
-2. Create branch if required
-3. Complete code changes and dev test
-4. Commit-Push changes in git Repository and Raise a pull request against Main Repository. Notify reviewer to review the changes.
-5. If any review comments come, perform step 3-4 again and resolve conversation. Same pull request will pick the changes.
```
Comment (vaibhavjain3) [inline CodingGuidelines.md:6]: why this file has been changed.
two points got missed from new file.

File: src/main/java/com/erstudio/ERStudioReleaseHelper.java [hunk 12]
```diff
@@ -13,10 +13,8 @@
      * Launches ERStudio Helper Application
      * @param args
      */
-
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERStudioReleaseHelper.java]: Why this line has been removed?

File: src/main/java/com/erstudio/ERStudioReleaseHelper.java [hunk 13]
```diff
@@ -13,10 +13,8 @@
...
     public static void main(String[] args) {
         // TODO Auto-generated method stub
         ERSStudioReleaseHelperUI ERStudioReleaseHelperUI = new ERSStudioReleaseHelperUI();
     }
-
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERStudioReleaseHelper.java:21]: Why this line has been removed?

File: src/main/java/com/erstudio/constants/Constants.java [hunk 14]
```diff
@@ -1,12 +1,11 @@
-package main.java.com.erstudio;
+package main.java.com.erstudio.constants;
 
 /**
  * @author akshit.arora
  * for String Constants
  */
 
 public class Constants {
-
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/constants/Constants.java]: Why this line has been removed?

File: src/main/java/com/erstudio/tsversionchange/ProcessExcel.java [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+                        excelModel.setPatchVersion(dataFormatter.formatCellValue(cell4));
+                        // string to replace format
+                        Cell cell5 = cellIterator.next();
+                        VersionFormat versionFormat = new VersionFormat(dataFormatter.formatCellValue(cell5));
+                        excelModel.setVersionFormat(versionFormat);
+                        fileList.add(excelModel);
+                    }
+                }
+            });
+            // workbook.close();
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/tsversionchange/ProcessExcel.java]: why this comment is here? Aren't we doing workbook.close in finally block already?

File: src/main/java/com/erstudio/tsversionchange/ProcessExcel.java [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+            try {
+                workbook.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        return fileList;
+    }
+}
+
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/tsversionchange/ProcessExcel.java]: What is this extra line at the end?

**Review by vaibhavjain3 — CHANGES_REQUESTED** [review]
Please review.

**Review by vaibhavjain3 — APPROVED** [review]
Approved

**Pull request merged**